src/renderer/src/utils/__tests__/mcp-tools.test.ts

```diff
@@ -0,0 +1,451 @@
+import { describe, expect, it } from 'vitest'
+
+// Local copy of filterProperties function to avoid import dependency issues in test environment
+function filterProperties(schema: any): any {
+  if (!schema || typeof schema !== 'object') {
+    return schema
+  }
```

Comment:
plz avoid keeping a local copy of this function in the test file. It's hard to maintain.

It's better to mock dependencies of `mcp-tools` or just move the function `filterProperties` to another lightweight util file, e.g., `mcp-schema`.